### ISSUE TITLE
Feature: CONTRIBUTING guidelines

### ISSUE THREAD

**Issue by omrishiv** [cnoe-io/stacks#8]
### Have you searched for this feature request?

- [X] I searched but did not find similar requests

### Problem Statement

As a newcomer (or a regular committer) it would be great to have contribution guidelines. A set of guidelines for how to structure the new stacks and maybe some example anti-patterns to stay away from would be great. 

### Possible Solution

Create a `CONTRIBUTING.md` with some guidance

**Comment by nabuskey** [comment]
Some examples for what we want in guidelines:

1. Version pinning if remote helm repos are used. 
2. When to use helm chart vs manifests.
3. Application Sets vs Applications.
4. Readme essentials:
    - Overview: What does this do? Why should you care?
    - Optional architecture diagram. 
    - Relevant links for further reading.

**Comment by cmoulliard** [comment]
I fully agree to have this guide as typically what we are suggesting here has not been implemented for by example the local stack ref implementation. 

That means also that we need a PR review process where we check/control if the needed information are well present
